WEB/Src/PerformanceCollector/Perf.Shared.NetFull/Implementation/QuickPulse/QuickPulseServiceClient.cs

```diff
@@ -381,6 +386,12 @@ private void AddHeaders(HttpWebRequest request, bool includeIdentityHeaders, str
                 request.Headers.Add(QuickPulseConstants.XMsQpsInvariantVersionHeaderName,
                     MonitoringDataPoint.CurrentInvariantVersion.ToString(CultureInfo.InvariantCulture));
             }
+
+            // The AAD token is an optional feature. Only include if it is provided.
+            if (!string.IsNullOrEmpty(aadToken))
```

Comment:
is this intentionally optional for QP? elsewhere, in the regular channel, its like mandatory, if user has set tc.settokencred.

Reply:
oh good catch. i need to rethink this